.github/workflows/pr-go-workspace-check.yml

```diff
@@ -0,0 +1,39 @@
+name: "Go Workspace Check"
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [main]
+    paths:
+      - '**/go.work'
+      - '**/go.work.sum'
+      - '**/go.mod'
+      - '**/go.sum'
+
+jobs:
+  check:
+    name: Go Workspace Check
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set go version
+      uses: actions/setup-go@v4
+      with:
+        go-version: '1.21.8'
+
+    - name: Build go files
+      run: |
+        make build-go
+
+    - name: Check for go workspace changes 
+      run: |
+        if ! git diff --exit-code go.work.sum; then
+          echo "There are changes in the go workspace"
```

Comment:
I think it would be nice if this suggest what to do next.

Reply:
updated in 1aae5eb541e123cbd687333b024fe75265bf8d14